docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+Process Overview: 
+1. The user issues a command to search for a specific expense i.e. `search expense Japan`. BudgetBuddy parses this
+input with the help of the `Parser` class.
+2. The `Parser` calls the `isCommand()` method of the `SearchExpenseCommand` class, to check if the user input 
+starts with "search expense". 
+3. If the user input starts with "search expense", the `Parser` then calls the `processCommand()` method of 
+a helper class, `SearchExpenseValidator` to extract the description to be filtered on.
+4. The `processCommand()` method above returns a new `SearchExpenseCommand` object initialized with the description
+extracted as the `keyword` attribute.
+5. BudgetBuddy then calls the `execute()` method of the `SearchExpenseCommand` object. 
+6. If the keyword attribute is an empty string, the `SearchExpenseCommand` object calls the `searchEmptyMessage()`
+method of the `Ui` class, displaying an error message to the user that no descriptor was provided.
+7. Else, the `SearchExpenseCommand` object calls the `searchExpenses()` method of the `ExpenseManager` class,
+filtering the `expenses` ArrayList and returning a String containing all expenses with the given descriptor
+in the description of the expenses. The `SearchExpenseCommand` object then calls the `displayToUser()` method in `Ui`,
+displaying this String to the user.
+
+#### 4.6 Display Savings Feature
+The Display Savings Feature enables users to check how much they have saved, through their inputs into the application.
+We assume that the user has accurately reflected all expenses and incomes, and we calculate their savings by 
+taking Savings = Total Income - Total Expense. The user has the option to either
+display their total savings, or their savings per month since using the app. This feature is managed by the 
+`DisplaySavingsCommand` class, initialized by the `Parser` class using the `DisplaySavingsValidator` class to display
+the correct savings, according to the user input. 
+The `DisplaySavingsCommand` object is then created with a boolean as an attribute. The class attributes and their
+relevance is as follows: 
+
+|Variable Name| Variable Type | Relevance                                                                   | 
+|-------------|---------------|-----------------------------------------------------------------------------|
+|byMonth| boolean | Indicates whether the user wants to display by month or just total savings. |
+
+The BudgetBuddy class then calls the `execute` method of the `DisplaySavingsCommand` object which uses the 
+`displaySavings()` or `displaySavingsByMonth()` method in the `SavingsManager` class, displaying the result to the 
+user using the `Ui` class `displayToUser()` method. 
+
+Below is a sequence diagram representing the execution of the Display Expense interaction:
+![DisplaySavingsSequenceDiagram.drawio.png](diagrams/DisplaySavingsSequenceDiagram.drawio.png)
```

Comment:
Should the different frames overlap like this? Also I believe again the alt and opt might have been swapped.
![image](https://github.com/user-attachments/assets/5c7e8544-20b6-4e79-b3ee-d73375990f85)